src/tools/tidy/src/deps.rs

```diff
@@ -361,6 +362,8 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "rand_xorshift", // dependency for doc-tests in rustc_thread_pool
     "rand_xoshiro",
     "redox_syscall",
+    "ref-cast",
```

Comment:
Remark: at least this uses rustc version detection, not just nightly detection

https://github.com/dtolnay/ref-cast/blob/1.0.24/build.rs